pytorch_lightning/metrics/classification/helpers.py

```diff
@@ -216,29 +222,30 @@ def _check_classification_inputs(
     When ``num_classes`` is not specified in these cases, consistency of the highest target
     value against ``C`` dimension is checked for (multi-dimensional) multi-class cases.
 
-    If ``top_k`` is set (not None) for inputs which are not (multi-dimensional) multi class
-    with probabilities, then an error is raised. Similarly if ``top_k`` is set to a number
-    that is higher than or equal to the ``C`` dimension of ``preds``.
+    If ``top_k`` is set (not None) for inputs which do not have probability predictions (and
+    are not binary), then an error is raised. Similarly if ``top_k`` is set to a number that
+    is higher than or equal to the ``C`` dimension of ``preds``.
 
     Preds and target tensors are expected to be squeezed already - all dimensions should be
-    greater than 1, except perhaps the first one (N).
+    greater than 1, except perhaps the first one (``N``).
 
     Args:
         preds: Tensor with predictions (labels or probabilities)
         target: Tensor with ground truth labels, always integers (labels)
         threshold:
             Threshold probability value for transforming probability predictions to binary
-            (0,1) predictions, in the case of binary or multi-label inputs. Default: 0.5
+            (0,1) predictions, in the case of binary or multi-label inputs.
         num_classes:
             Number of classes. If not explicitly set, the number of classes will be infered
             either from the shape of inputs, or the maximum label in the ``target`` and ``preds``
             tensor, where applicable.
         top_k:
             Number of highest probability entries for each sample to convert to 1s - relevant
-            only for (multi-dimensional) multi-class inputs with probability predictions. The
-            default value (``None``) will be interepreted as 1 for these inputs.
+            only for inputs with probability predictions. The default value (``None``) will be
+            interepreted as 1 for these inputs. If this parameter is set for multi-label inputs,
+            it will take precedence over threshold.
 
-            Should be left unset (``None``) for all other types of inputs.
+            Should be left unset (``None``) for inputs with label predictions.
         is_multiclass:
```

Comment:
did you forget to update is_multiclass docs here, just like you did at other places using 
```
:ref:`documentation section <metrics:Using the \\`\\`is_multiclass\\`\\` parameter>`
```